fairworkflows/rdf_wrapper.py

```diff
@@ -28,14 +29,38 @@ def is_modified(self) -> bool:
         """Returns true if the RDF has been modified since initialisation"""
         return self._is_modified
 
+    def remove_non_attribute_triples(self, keep_triples: List[Tuple] = None):
```

Comment:
This is something I've been meaning to get to for a while, and should have created an issue for. Initially we indeed just look for triples that have the step/workflow URI as the subject, but it could be a bit trickier than that. There can be 'non-attribute' triples that describe an object of an attribute triple. e.g. we might say that the step (```baseURI#step```) uses a piece of lab equipment, ```baseURI#equipment```, and then there could be several triples of the form ```(baseURI#equipment, predicate, object)``` and then even more triples describing one of the objects of these triples. But all of these 'non-attribute' triples are actually very much part of the step description. (I hope what I'm saying makes some sort of sense).

I think we may need to 'walk' the graph a little to make sure all triples relating to this step are included. But we also want to avoid pulling in the stuff from plex workflows, such as precedes relations, since these are not intrinsic to the step definition. I suppose we could exclude any triples found during our walk that contain workflow related predicates? Not sure if it is that trivial or if we need to give it more though.

Reply:
Yeah, good point. This is too easy. Let's discuss!

Reply:
I worked out a solution that is more advanced.